modules/dev-utils/src/main/java/org/apache/ignite/development/utils/indexreader/FilePageStoreFactory.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.development.utils.indexreader;
+
+import java.io.File;
+import java.nio.ByteBuffer;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.pagemem.PageIdAllocator;
+import org.apache.ignite.internal.processors.cache.persistence.file.FilePageStore;
+import org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager;
+import org.jetbrains.annotations.Nullable;
+
+import static java.lang.String.format;
+import static java.util.Objects.nonNull;
+import static org.apache.ignite.internal.pagemem.PageIdAllocator.INDEX_PARTITION;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.FILE_SUFFIX;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.INDEX_FILE_NAME;
+import static org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreManager.PART_FILE_TEMPLATE;
+
+/**
+ * Factory {@link FilePageStore} for analyzing partition and index files.
+ */
+public interface FilePageStoreFactory {
```

Comment:
Please rename this interface as it's name is in conflict with `org.apache.ignite.internal.processors.cache.persistence.file.FilePageStoreFactory`. I believe it's possible because this FilePageStoreFactory is used only for needs of index reader.

Reply:
IgniteIndexReaderFilePageStoreFactory - ok?